algorithms/linfa-pls/src/lib.rs

```diff
@@ -157,12 +157,18 @@ macro_rules! pls_algo { ($name:ident) => {
             }
         }
 
-        impl<F: Float, D: Data<Elem = F>> PredictRef<ArrayBase<D, Ix2>, Array2<F>> for [<Pls $name>]<F> {
+        impl<F: Float, D: Data<Elem = F>> PredictInplace<ArrayBase<D, Ix2>, Array2<F>> for [<Pls $name>]<F> {
             /// Given an input matrix `X`, with shape `(n_samples, n_features)`,
             /// `predict` returns the target variable according to [<Pls $name>] method
             /// learned from the training data distribution.
-            fn predict_ref<'a>(&'a self, x: &ArrayBase<D, Ix2>) -> Array2<F> {
-                self.0.predict_ref(x)
+            fn predict_inplace<'a>(&'a self, x: &ArrayBase<D, Ix2>, y: &mut Array2<F>) {
+                assert_eq!(x.nrows(), y.nrows(), "The number of data points must match the number of output targets.");
+
+                self.0.predict_inplace(x, y);
+            }
+
+            fn num_targets(&self) -> usize {
+                self.0.num_targets()
```

Comment:
I'm not sure where the trait bound is missing :confused:

Reply:
For `pls_generic.rs` `self.y_mean` is an `Array1` though so it doesn't have `ncols()`. Maybe `self.coefficients.ncols()`?

Reply:
The compiler didn't know which version of `PredictInplace` we wanted to call, because unlike in `predict_inplace` the generic target type isn't implied.